right/src/secondary_role_driver.c

```diff
@@ -132,18 +135,26 @@ static secondary_role_state_t resolveCurrentKeyRoleIfDontKnowTimeout()
         && resolutionKey == previousResolutionKey
         && resolutionStartTime - previousResolutionTime < SecondaryRoles_AdvancedStrategyDoubletapTime
         ) {
+        KEY_TIMING(KeyTiming_RecordComment(resolutionKey, "PA"));
```

Comment:
This function is becoming quite heavy and difficult to comprehend. Is there any way to refactor it to multiple functions?

Reply:
I would much rather not do that.

The function is long, but straightforward. Chopping it into multiple function would just make it much harder to navigate.

Reply:
Fair enough.